### PR TITLE
[MLOB-1696] Add Node.js LLM Observability SDK docs

### PR DESCRIPTION
**Pull request by sabrenner**
### What does this PR do? What is the motivation?
Adds Node.js SDK and auto-instrumentation documentation for LLM Observability. It also configures the `_index.md` file for our `sdk` directory to be a select based on language, since we plan on adding more language SDKs down the line.

### Merge instructions
Will be waiting on version `5.25.0` of `dd-trace-js`. I'll ping the documentation team when it is ready to be merged.

### CODE REVIEW

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:28]: Are we going to be making users install off a branch? Or are we going to merge the SDK before we change/publish this?

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:132]:
```suggestion
### Automatic function argument/output/name capturing
```

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:350]: Small nit (realized it's the same issue with the python page): can we add 1 line here explaining this also includes the annotation example? (and same for the python page)

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:371]: Can we link to `tracer.wrap()` documentation here for people that may not be APM customers?

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:377]: Which case is the below example based on here?

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:434]: Question (not doc related): does the Node SDK annotate the current active LLMOBS-type span? Or would this conflict with APM/LLMOBS type spans?

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:602]: Clarifying question - is this still accurate for `llmobs.trace()`? I thought inline tracing methods automatically finish the span once you exit the block

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:604]: What does this mean here?

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:621]:
```suggestion
1. Start a span manually using `llmobs.startSpan(spanKind, options)`
```

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:619]: Might be worth introducing `llmobs.startSpan()` as a tool provided by the SDK to manually start and stop a span here.

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [inline content/en/llm_observability/setup/sdk/nodejs.md:134]:
```suggestion
`llmobs.wrap` (along with [`llmobs.decorate`](#function-decorators-in-typescript) for typescript) will try to automatically capure inputs, outputs, and the name of the function being traced. If you need to manually annotate a span, see [Annotating a span](#annotating-a-span). Inputs and outputs you annotate will override the automatic capturing. Additionally, to override the function name, pass the `name` property on the options object to the `llmobs.wrap` function:
```
Should mention typescript here, otherwise people might get a bit confused by the second function name dropping

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/nodejs.md:28]: This is just here now for initial use from design partners. When I come back to give this PR a final pass once the SDK is released, I'll change this to
```
npm install dd-trace
```

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/nodejs.md:434]: yeah it would conflict if the current span was not started by the SDK/LLMObs integration. in that case, it logs and returns early.

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/nodejs.md:602]: it's the same here, i think for the underlying tracer it's about consistency. so, if the function returns a promise, the span will finish when the promise is resolved or rejected, not when the promise is returned.

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/nodejs.md:604]: yeah this wording is iffy - if the underlying function returns an int, for example, `llmobs.trace` will have that return type as well, eg:
```javascript
const a = llmobs.trace('llm', () => {
  return 5
});
console.log(a); // logs 5

const obj = llmobs.trace('llm', () => {
  return { a: 5 }
});
console.log(obj); // logs { a: 5 }
```

Do those examples help clarify? I can add an example like this to aide this definition if so! Otherwise, happy to reword it. maybe i should explicitly mention "type" (ie return type)

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/nodejs.md:619]: could you clarify what you mean by this? I have an example below using `llmobs.startSpan`, is that what you're referring to? for reference, i just followed this section: https://docs.datadoghq.com/llm_observability/setup/sdk/#persisting-a-span-across-contexts

Comment (Yun-Kim) [reply content/en/llm_observability/setup/sdk/nodejs.md:619]: I'm just pointing out that users might find it confusing when the first mention of `llmobs.startSpan()` is in step 1 of the below example without any description of what it is/does (I know, should be obvious but it always helps to be 100% clear in these cases). Maybe something along the lines of:
```
The LLM Observability Node.js SDK provides the `llmobs.startSpan()` method to manually start and stop a span to be persisted across different contexts or scopes:
```

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [reply content/en/llm_observability/setup/sdk/nodejs.md:604]: Oh I see. Maybe a better wording is:
```The return type of this function will match the return type of the wrapped function.
```

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [reply content/en/llm_observability/setup/sdk/nodejs.md:434]: Cool, we'll talk more about context management next week to ensure we'll always find the current active LLMObs span instead of this current behavior

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/nodejs.md:619]: ahh ok yes that makes sense, will add something like that!

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Yun-Kim) [reply content/en/llm_observability/setup/sdk/nodejs.md:377]: No problem, maybe just worth noting that the example is reflective of case 2 for users that might get confused.

**Comment by neko-dd** [comment]
Created DOCS-9395 for docs team review

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:12]:
```suggestion
<div class="alert alert-warning">LLM Observability is not available in the selected site ({{< region-param key="dd_site_name" >}}).</div>
```

File: content/en/llm_observability/setup/auto_instrumentation.md
Comment (Kyle-Verhoog) [inline content/en/llm_observability/setup/auto_instrumentation.md:17]: :+1: nice - i like the tab approach. I would propose taking it a step further by just having the code-snippets be language specific and leave the bits in between (like definitions, intros to the product terms) to be shared. 

I can definitely see the duplicated info between the Python and Node.js pages going out of sync real fast.

I also realize this is a big refactor to these docs so I'll leave it up to you @sabrenner

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:407]:
```suggestion
<br />A dictionary of JSON serializable key-value pairs that users can add as metadata information relevant to the input or output operation described by the span (`model_temperature`, `max_tokens`, `top_k`, etc.).
```

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:411]:
```suggestion
<br />A dictionary of JSON serializable keys and numeric values that users can add as metrics relevant to the operation described by the span (`input_tokens`, `output_tokens`, `total_tokens`, `time_to_first_token`, etc.). The unit for `time_to_first_token` is in seconds, similar to the `duration` metric which is emitted by default.
```

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:415]:
```suggestion
<br />A dictionary of JSON serializable key-value pairs that users can add as tags regarding the span's context (`session`, `environment`, `system`, `versioning`, etc.). For more information about tags, see [Getting Started with Tags][9].
```

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:519]:
```suggestion
<br />The type of the evaluation. Must be `categorical` or `score`.
```

File: content/en/llm_observability/setup/auto_instrumentation.md
Comment (Kyle-Verhoog) [inline content/en/llm_observability/setup/auto_instrumentation.md:216]: IMO it's overkill for us to list out each method we instrument. I don't think customers will care much and just want it to work :tm:

Given that this is already a pattern I'm ok to leave as-is and we can go back over everything and omit them in a follow up.

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:523]:
```suggestion
<br />The value of the evaluation. Must be a string (`metric_type==categorical`) or integer/float (`metric_type==score`).
```

File: content/en/llm_observability/setup/auto_instrumentation.md
Comment (Kyle-Verhoog) [inline content/en/llm_observability/setup/auto_instrumentation.md:194]: It looks like Python uses `<INTEGRATION_NAME_IN_LOWERCASE>` but I love putting real values in examples that actually work and require less grokking of the example. Also I don't like the implication that the user has to go find out the right magic string to make this work.

```suggestion
tracer.use('openai', true);
tracer.use('langchain', true);
// etc
```

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:553]:
```suggestion
For each span kind, the `ddtrace.llmobs.LLMObs` class provides a corresponding inline method to automatically trace the operation a given code block entails. These methods have the same argument signature as their function decorator counterparts, with the addition that `name` defaults to the span kind (`llm`, `workflow`, etc.) if not provided. These methods can be used as context managers to automatically finish the span after the enclosed code block is completed.
```

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:572]:
```suggestion
3. Stop the span manually with the `span.finish()` method. **Note**: the span must be manually finished, otherwise it is not submitted.
```

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (Kyle-Verhoog) [inline content/en/llm_observability/setup/sdk/nodejs.md:50]: this is awesome that we have the site here. Can we also put it in the codeblock to auto-fill it in? One less step for the user.

File: content/en/llm_observability/setup/sdk/python.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/python.md:597]: Heading says "tracing," first sentence says "tracking"—is one of these an error?

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (cswatt) [inline content/en/llm_observability/setup/sdk/nodejs.md:678]: Heading says "tracing", first sentence says "tracking" — which one is correct?

**Review by Kyle-Verhoog — COMMENTED** [review]
overall looks good and is consistent with the Python docs :+1:

Just one actionable comment, rest are optional or up to you!

Great job Sam! 👏

File: content/en/llm_observability/setup/sdk/python.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/python.md:597]: should probably be "tracing" - will update!

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/nodejs.md:678]: should probably be "tracing" - will update!

File: content/en/llm_observability/setup/auto_instrumentation.md
Comment (cswatt) [reply content/en/llm_observability/setup/auto_instrumentation.md:17]: We can merge as-is for now, and I can take on that refactor in a later PR!

File: content/en/llm_observability/setup/auto_instrumentation.md
Comment (sabrenner) [reply content/en/llm_observability/setup/auto_instrumentation.md:212]: yeah that makes sense to me. happy to follow up with it separately so we're consistent for python and node.js

File: content/en/llm_observability/setup/auto_instrumentation.md
Comment (sabrenner) [reply content/en/llm_observability/setup/auto_instrumentation.md:190]: i like that too! will change for both node.js and python

File: content/en/llm_observability/setup/sdk/nodejs.md
Comment (sabrenner) [reply content/en/llm_observability/setup/sdk/nodejs.md:50]: yes, good idea!

File: content/en/llm_observability/setup/auto_instrumentation.md
Comment (sabrenner) [reply content/en/llm_observability/setup/auto_instrumentation.md:17]: thanks @cswatt, really appreciate that! i'm happy to help out with that when/if needed 😄

**Review by cswatt — APPROVED** [review]
This all looks good! Will be ready to go after build pipeline passes. @sabrenner when would you like to merge?

**Comment by sabrenner** [comment]
@cswatt Thanks! I'll ping you directly, we're still waiting on the actual release to happen from `dd-trace-js`, once it's release we're good to go to merge. Should be this afternoon, but I'll keep you posted

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by davidejones — APPROVED** [review]

**Pull request merged**
